lib/puppet/indirector/rest.rb

```diff
@@ -277,6 +281,29 @@ def parse_response(response)
     end
   end
 
+  # This is a copy of convert_to_http_error above, but adapted to be
+  # used with Puppet::HTTP::ResponseErrors
+  def handle_response_error(url, request, response)
+    if is_http_404?(response)
+      return nil unless request.options[:fail_on_404]
+
+      _, body = parse_response(response)
+      msg = _("Find %{uri} resulted in 404 with the message: %{body}") % { uri: elide(url.path.to_s, 100), body: body }
+      raise Puppet::Error, msg
+    else
+      raise convert_to_http_error(response)
+    end
+  end
+  protected :handle_response_error
+
+  # This is a copy of parse_response, but it raises Puppet::Error instead
+  def content_type(response)
```

Comment:
It doesn't look like this is called anywhere?

Reply:
Removed